src/App.tsx

```diff
@@ -15,49 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  SORT_BY_ALPHABET = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+}
+
+function sortedGoods(
+  goods: string[],
+  selectedOrder: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (selectedOrder) {
+    visibleGoods.sort((first: string, second: string) => {
+      switch (selectedOrder) {
+        case SortType.SORT_BY_ALPHABET:
+          return first.localeCompare(second);
+
+        case SortType.SORT_BY_LENGTH:
+          return first.length - second.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [selectedOrder, setSelectedOrder] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = sortedGoods(goodsFromServer, selectedOrder, isReversed);
+
+  const resetButton = () => {
```

Comment:
```suggestion
  const handleReset = () => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364